python_scripts/ensemble_gradient_boosting.py

```diff
@@ -8,26 +8,28 @@
 # %% [markdown]
 # # Gradient-boosting decision tree (GBDT)
 #
-# In this notebook, we will present the gradient boosting decision tree
-# algorithm and contrast it with AdaBoost.
+# In this notebook, we present the gradient boosting decision tree algorithm.
 #
-# Gradient-boosting differs from AdaBoost due to the following reason: instead
-# of assigning weights to specific samples, GBDT will fit a decision tree on the
-# residuals error (hence the name "gradient") of the previous tree. Therefore,
-# each new tree in the ensemble predicts the error made by the previous learner
-# instead of predicting the target directly.
+# Even if AdaBoost and GBDT are both boosting algorithms, they are different in
+# nature: the former assigns weights to specific samples, whereas GBDT fits
+# succesive decision trees on the residual errors (hence the name "gradient") of
+# their preceding tree. Therefore, each new tree in the ensemble tries to refine
+# its predictions by specifically addressing the errors made by the previous
+# learner, instead of predicting the target directly.
 #
-# In this section, we will provide some intuition about the way learners are
-# combined to give the final prediction. In this regard, let's go back to our
-# regression problem which is more intuitive for demonstrating the underlying
+# In this section, we provide some intuitions on the way learners are combined
+# to give the final prediction. For such purpose, we tackle a single-feature
+# regression problem, which is more intuitive for demonstrating the underlying
 # machinery.
+#
+# Later in this notebook we compare the performance of GBDT (boosting) with that
+# of a Random Forest (bagging) for a particular dataset.
 
 # %%
 import pandas as pd
 import numpy as np
 
-# Create a random number generator that will be used to set the randomness
-rng = np.random.RandomState(0)
+rng = np.random.RandomState(0)  # Create a random number generator
```

Comment:
Let's move the generator next to the data generation. We should avoid showing a pattern where people would use the generator across different function and estimators.
So the best practice is just to show it next to the data generation. we could even slightly change the code and have:

```python
def generate_data(n_samples=50, seed=0):
    rng = np.random.default_rng(seed)
    x = rng.normal(size=(n_samples,)) * ...
    noise = rng.normal(size=(s_samples,)) * 0.3
```

using `default_rng` should be encourage nowadays.